locked-nft/contracts/NFTLocker.cdc

```diff
@@ -89,27 +93,177 @@ access(all) contract NFTLocker {
+                authorizedDepositHandler: authorizedDepositHandler,
+                eligibleNFTTypes: eligibleNFTTypes
+            )
+
+            // Add the receiver to the lookup map
+            for nftType in eligibleNFTTypes.keys {
+                if let namesMap = self.receiverNamesByNFTType[nftType] {
+                    namesMap[name] = true
+                    self.receiverNamesByNFTType[nftType] = namesMap
+                } else {
+                    self.receiverNamesByNFTType[nftType] = {name: true}
+                }
+            }
+        }
+
+        /// Remove a deposit method for a given NFT type
+        ///
+        access(Operate) fun removeReceiver(name: String) {
+            // Get the receiver
+            let receiver = self.receiversByName[name]
+                ?? panic("Receiver with the given name does not exist")
+
+            // Remove the receiver from the lookup map
+            for nftType in receiver.eligibleNFTTypes.keys {
+                if self.receiverNamesByNFTType.containsKey(nftType) {
+                    self.receiverNamesByNFTType[nftType]!.remove(key: name)
+                }
+            }
+
+            // Remove the receiver
+            self.receiversByName.remove(key: name)
+        }
+
+        /// Get the deposit method for the given name if it exists
+        ///
+        access(all) view fun getReceiver(name: String): Receiver? {
+            return self.receiversByName[name]
+        }
+
+        /// Get the receiver names for the given NFT type if it exists
+        ///
+        access(all) view fun getReceiverNamesByNFTType(nftType: Type): {String: Bool}? {
+            return self.receiverNamesByNFTType[nftType]
+        }
+
+        /// Initialize ReceiverCollector struct
+        ///
+        view init() {
+            self.receiversByName = {}
+            self.receiverNamesByNFTType = {}
+            self.metadata = {}
+        }
+    }
+
     /// Admin resource
+    ///
     access(all) resource Admin {
+        /// Expire lock
+        ///
         access(all) fun expireLock(id: UInt64, nftType: Type) {
-            if let locker = &NFTLocker.lockedTokens[nftType] as auth(Mutate) &{UInt64: NFTLocker.LockedData}?{
-                if locker[id] != nil {
-                    // remove old locked data and insert new one with duration 0
-                    if let oldLockedData = locker.remove(key: id){
-                        let lockedData = NFTLocker.LockedData(
+            NFTLocker.expireLock(id: id, nftType: nftType)
+        }
+
+        /// Create and return a new ReceiverCollector resource
+        ///
+        access(all) fun createReceiverCollector(): @ReceiverCollector {
+            return <- create ReceiverCollector()
+        }
+    }
+
+    /// Expire lock if the locked NFT is eligible for force unlock and deposit by authorized receiver
+    ///
+    access(contract) fun expireLock(id: UInt64, nftType: Type) {
```

Comment:
`expireLock` definition is moved at the contract level (with `access(contract)`) in order to be called from `unlockWithAuthorizedDeposit`, if authorized